src/traversals/bfs.jl

```diff
@@ -1,98 +1,19 @@
-### DEVELOPERS NOTE: BFS optimization experiments are typically
-### prototyped in gdistances!, since it's one of the simplest
-### BFS implementations.
+#### REMOVE-2.0
 
-"""
-    tree(parents)
-
-Convert a parents array into a directed graph.
-"""
-function tree(parents::AbstractVector{T}) where T <: Integer
-    n = T(length(parents))
-    t = DiGraph{T}(n)
-    for (v, u) in enumerate(parents)
-        if u > zero(T)  && u != v
-            add_edge!(t, u, v)
-        end
-    end
-    return t
+function bfs_parents(g::AbstractGraph, s::Integer; dir = :out)
+    Base.depwarn("`bfs_parents` is deprecated. Equivalent functionality has been moved to `LightGraphs.Traversals.parents`.", :bfs_parents)
+    z = dir == :out ? LightGraphs.Traversals.parents(g, s, LightGraphs.Traversals.BreadthFirst()) : LightGraphs.Traversals.parents(g, s, LightGraphs.Traversals.BreadthFirst(neighborfn=inneighbors))
+    z[s] = s # fixing this in 2.0
+    return z
 end
 
-"""
-    bfs_parents(g, s[; dir=:out])
-
-Perform a breadth-first search of graph `g` starting from vertex `s`.
-Return a vector of parent vertices indexed by vertex. If `dir` is specified,
-use the corresponding edge direction (`:in` and `:out` are acceptable values).
-
-
-### Performance
-This implementation is designed to perform well on large graphs. There are
-implementations which are marginally faster in practice for smaller graphs,
-but the performance improvements using this implementation on large graphs
-can be significant.
-"""
-bfs_parents(g::AbstractGraph, s::Integer; dir = :out) = 
-    (dir == :out) ? _bfs_parents(g, s, outneighbors) : _bfs_parents(g, s, inneighbors)
-
-function _bfs_parents(g::AbstractGraph{T}, source, neighborfn::Function) where T
-    n = nv(g)
-    visited = falses(n)
-    parents = zeros(T, nv(g))
-    cur_level = Vector{T}()
-    sizehint!(cur_level, n)
-    next_level = Vector{T}()
-    sizehint!(next_level, n)
-    @inbounds for s in source
-        visited[s] = true
-        push!(cur_level, s)
-        parents[s] = s
-    end
-    while !isempty(cur_level)
-        @inbounds for v in cur_level
-            @inbounds @simd for i in  neighborfn(g, v)
-                if !visited[i]
-                    push!(next_level, i)
-                    parents[i] = v
-                    visited[i] = true
-                end
-            end
-        end
-        empty!(cur_level)
-        cur_level, next_level = next_level, cur_level
-        sort!(cur_level)
-    end
-    return parents
+function bfs_tree(g::AbstractGraph, s::Integer; dir = :out)
+    Base.depwarn("`bfs_tree` is deprecated. Equivalent functionality has been moved to `LightGraphs.Traversals.tree`.", :bfs_tree)
```

Comment:
why not use the `@deprecate` macro directly? in this file and the ones below?

Reply:
Because this isn't a like-for-like deprecation. The new `Traversals.tree` takes a `Traversals.BreadthFirst()` object as the second parameter to be equivalent, and we have the ability to do other things as well, so I didn't want to do a standard `@deprecate`.

Reply:
all good!